playlist.c

```diff
@@ -2769,18 +2780,47 @@ void playlist_get_db_name(playlist_t *playlist, size_t idx,
    }
 }
 
-char *playlist_get_default_core_path(playlist_t *playlist)
+core_info_t *playlist_entry_get_core(const struct playlist_entry* entry)
 {
-   if (!playlist)
+   core_info_ctx_find_t core_info;
+
+   core_info.inf = NULL;
+   core_info.path = entry->core_path;
+
+   if (!core_info_find(&core_info))
       return NULL;
-   return playlist->default_core_path;
+
+   if (strcmp(core_info.inf->path, entry->core_path))
+   {
+      RARCH_LOG("[CORE] Found core '%s' path '%s' does not match requested path '%s'", core_info.inf->display_name, core_info.inf->path, entry->core_path);
+   }
+
+   return core_info.inf;
 }
 
-char *playlist_get_default_core_name(playlist_t *playlist)
```

Comment:
You've removed the `playlist_get_default_core_name()` function - this breaks several things...